ocaml/xapi/message_forwarding.ml

```diff
@@ -3226,6 +3230,16 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
         with _ -> raise Not_found in
       do_op_on ~local_fn ~__context ~host op
 
+    (* Forward SR operation to all hosts that have suitable, plugged or
+       unplugged PBDs and the pool master *)
+    let forward_sr_all_op ~local_fn ~__context ~self op =
+      let module HostSet = Set.Make(struct type t = API.ref_host let compare = Pervasives.compare end) in
+      choose_pbds_for_sr ~consider_unplugged_pbds:true ~__context ~self ()
+      |> List.map (fun pbd -> Db.PBD.get_host ~__context ~self:pbd)
+      |> List.cons (Helpers.get_master ~__context)
```

Comment:
it looks like you are using a set just so you can cons the master? i think you should just do this instead:

```
let pbds_for_sr = if List.mem master pbds_for_sr then pbds_for_sr else master :: pbds_for_sr in
```

if `choose_pbds_for_sr` does indeed return duplicates, should it?

Reply:
I guess it's simpler that way, with equivalent cost.